### DISCUSSION TITLE
Is IPv6 supported in kernel 4.x?

### DISCUSSION THREAD

**Discussion by chbhunia** [strongswan/strongswan#1123] [Q&A]
Hi,

Can anyone please tell me if strongswan IPSec with IPv6 is supported in Kernel 4.x or not?
What is minimum kernel version needed to work IPv6 for IPSec?
I  am seeing some issue while running with the IPv6. Although the tunnels are established, the packets did not get encapsulated/encrypted.

Can anyone please help?

Thank you .

**Comment by tobiasbrunner** [comment]
Could definitely be that there were issues with older kernels, but IPv6 has definitely been supported since 2.6 kernels. See #196.

**Reply by chbhunia** [reply]
Tried in 4.15 kernel. 

We are facing this issue.

Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[KNL] received netlink error: Invalid argument (22)
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[KNL] unable to add SAD entry with SPI c3ddb812 (FAILED)
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[IKE] unable to install inbound and outbound IPsec SA (SAD) in kernel
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[IKE] failed to establish CHILD_SA, keeping IKE_SA
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[CHD] CHILD_SA ipsec-ikev2-vpn{1} state change: INSTALLING => DESTROYING
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[KNL] deleting policy 2001::/16 === 2001::/16 in
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[KNL] deleting policy 2001::/16 === 2001::/16 in failed, not found
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[KNL] deleting policy 2001::/16 === 2001::/16 fwd
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[KNL] deleting policy 2001::/16 === 2001::/16 fwd failed, not found
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[KNL] deleting SAD entry with SPI c2b1f095
Jul 19 05:14:54 5GCTE-1f5861d60c charon[3659]: 15[KNL] sending XFRM_MSG_DELSA 206: => 40 bytes @ 0x7fcd35b066f0


Do we need to disable AES-NI?

Here is the config:
 conn ipsec-ikev2-vpn
   left=1::2
   leftsubnet=2001::2/16
   leftid=1::2
   leftcert=server.cert.pem
   right=1::3
   rightsubnet=2001::3/16
   rightauth=eap-mschapv2
   rightid=%any
   eap_identity=%identity
   ike=aes256gcm16-sha384-prfsha384-ecp384!
   esp=aes256-sha256!
   dpddelay=300
   dpdtimeout=150
   dpdaction=1
   sha256_96=no
   mediation=no
   keyexchange=ikev2

**Reply by tobiasbrunner** [reply]
Make sure your kernel has the [required modules](https://docs.strongswan.org/docs/5.9/install/kernelModules.html) loaded (including crypto modules). Also note that you need at least a 5.8 kernel for UDP encapsulation for IPv6, so make sure no UDP encap is negotiated between the peers, or update to a newer kernel.
